src/cli/platform/caching.rs

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::path::Path;
+
+use sha2::{Digest, Sha256};
+
+/// Fetch a URL and save the contents as the hash of the URL
+pub fn save_file(url: &str, dest: &Path) -> anyhow::Result<()> {
+    let response = ureq::get(url).call()?.into_reader();
+    let response: Vec<u8> = response.bytes().map(|d| d.unwrap()).collect();
```

Comment:
Scratch this, and use std::io:: copy() instead of write_all below.

Reply:
How does the `ureq` response get turned into a reader for `copy()`? I'm getting an error that `std::io::Read` is not implemented for `Vec<u8>`, and `ureq::get(url).call()?.into_reader()` isn't helpful either.

Why is `copy()` desirable over `file.write_all()`?

Reply:
Don't use the Vec, use the into_reader() return value for io::copy

Reply:
@haraldh